# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/AudioConduit.cpp
# Commit: d3eb7bddf958
# Full Hash: d3eb7bddf9586c2ed7144ce5e9afef706342a085
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 1
# Description:
#   Bug 1654112 - All GetAudioFrame paths must unlock the locked mutex. r=padenot
#   
#   Unclear why this path was missed in the original patch.
#   Hopefully I didn't outsmart myself.
#   
# ==============================================================================

diff -r 370586931995 -r d3eb7bddf958 dom/media/webrtc/libwebrtcglue/AudioConduit.cpp
--- a/dom/media/webrtc/libwebrtcglue/AudioConduit.cpp	Fri Aug 20 11:16:39 2021 +0200
+++ b/dom/media/webrtc/libwebrtcglue/AudioConduit.cpp	Fri Aug 20 11:18:55 2021 +0200
@@ -463,13 +463,13 @@
   auto info = static_cast<webrtc::internal::AudioReceiveStream*>(mRecvStream)
                   ->GetAudioFrameWithInfo(samplingFreqHz, frame);
 
+  mMutex.Unlock();
+
   if (info == webrtc::AudioMixer::Source::AudioFrameInfo::kError) {
     CSFLogError(LOGTAG, "%s Getting audio frame failed", __FUNCTION__);
     return kMediaConduitPlayoutError;
   }
 
-  mMutex.Unlock();
-
   // Spec says to "queue a task" to update contributing/synchronization source
   // stats; that's what we're doing here.
   mCallThread->Dispatch(NS_NewRunnableFunction(
